packages/veritone-react-common/src/components/SentimentEngineOutput/index.js

```diff
@@ -1,189 +1,244 @@
+      timeWindowStartMs,
+      timeTickIntervalMs
+    } = this.props;
+
+    let chartData = [{sartTimeMs: 0, stopTimeMs: 0, sentiment: 0}];
+    let numValidValue = 0;
+    let totalSentiment = 0;
+    if (data && Array.isArray(data) && data.length > 0) {
+      data.map((entry, index) => {
+        if(!entry.hasOwnProperty('status') || entry.status === 'complete') {
+          let sentimentValue = 0;
+          let sentiment = entry.sentiment;
+          if (sentiment.positiveConfidence > sentiment.negativeConfidence) {
+            sentimentValue = sentiment.positiveValue * 100;
+          } else if (sentiment.positiveConfidence < sentiment.negativeConfidence) {
+            sentimentValue = sentiment.negativeValue * 100;
+          }
+  
+          numValidValue++;
+          totalSentiment = totalSentiment + sentimentValue;
+  
+          let chartFriendlyData = {
+            startTimeMs: entry.startTimeMs,
+            stopTimeMs: entry.stopTimeMs,
+            sentiment: sentimentValue
+          }
+  
+          chartData.push(chartFriendlyData);
+        } else {
+          //handle failed & pending status
+          chartData.push({
+            startTimeMs: entry.startTimeMs,
+            stopTimeMs: entry.stopTimeMs,
+            sentiment: 0
+          });
         }
-        normalizedSeries.splice(j + k, 0, {
-          score: 0,
-          time: time
-        });
-      }
+      });
+    }
+    
+    let totalTime = data[data.length - 1].stopTimeMs;
+    let xDomain = [0, totalTime];
+    let xTicks = [];
+    for (let xTickValue = 0; xTickValue <= totalTime; xTickValue = xTickValue + timeTickIntervalMs) {
+      xTicks.push(xTickValue);
     }
-    paddingInterval = 5000;
+
+    return {
+      chartData: chartData,
+      average: totalSentiment / numValidValue,
+      xTicks: xTicks,
+      xDomain: xDomain,
+      yTicks: sentimentTicks,
+      yDomain: sentimentDomain,
+      referenceValue: mediaPlayerTime,
+      totalTime: totalTime,
+      scrollToTime: timeWindowStartMs,
+      scaleX: totalTime / timeWindowSizeMs
+    };
   }
 
-  // Calculate average of adjusted data (ignoring gaps with no sentiment data):
-  let average = !seriesLength ? 0 : sumOfScores / seriesLength;
-  // Round average to 1 decimal place:
-  average = Math.round(average * 10) / 10;
+  setScrollRef = (target) => {
+    this.scrollContent = target;
+  }
 
-  return {
-    series: normalizedSeries,
-    average: average
-  };
+  scrollTo (timeMs, totalTimeMs) {
+    let scrollTarget = this.scrollContent;
+    let scrollRatio = timeMs / totalTimeMs;
+    
+    scrollTarget.scrollLeft = scrollRatio * scrollTarget.scrollWidth;
+  }
 
-  // Map "score" from [0,1] range to [-100,100] range and set "time" to midpoint of start/end:
-  function normalizeDatum(datum) {
-    let normalizedDatum = {
-      score: (datum.score - 0.5) * 200,
-      time: (datum.start + datum.end) / 2
-    };
+  formatXValues (value) {
+    return msToReadableString(value);
```

Comment:
date lib doesn't give us the values that we want (0 hours & days breaks it); therefore, we have to use our custom function to handle time format.